lib/dialects/mssql/query.js

```diff
@@ -112,6 +110,10 @@ class Query extends AbstractQuery {
 
       connection.execSql(request);
     });
+
+    complete();
```

Comment:
This means, benchmark will now include Sequelize record parsing overhead.

Earlier `benchmark = time to execute query by dialect`
Now `benchmark = time to execute query by dialect + time taken by sequelize to format result`

This deviation can be significant when parsing large number of records.

Main question is why placing `complete()` here works, do you have any ideas @Mikxail

Reply:
Actually, I don't know how `cls-hooked` works under the hood. But I can assume that the problem in `connection.lib.Request` that saves `callback` function in variable, and then call `callback` from `execSql`. Probably that can breaks `cls-hooked` algorithms.
A similar situation was in the `run` function (dialects/mysql/query.js).

I can refactor `_run` function (dialects/mssql/query.js), but it require a lot of changes. Can I do it withing this MR or I should create separate MR?

Reply:
I think issues listed below are variant of this issue
- https://github.com/nodejs/node/issues/21078
- https://github.com/Jeff-Lewis/cls-hooked/issues/36

At Sequelize level, we can extract record/error formatting away from callback. Something like this should work

1. Get raw result/error from callback where `triggerAsyncId` is incorrect (new Promise wrapper can be used)
2. Call `complete`
3. Parse error or raw results and return final result.

Reply:
I agree with it.
I did refactoring `_run` functions in dialects/mssql.